installer/src/panels/com/izforge/izpack/installer/AutomatedInstaller.java

```diff
@@ -108,7 +125,49 @@ public AutomatedInstaller(String inputFilename) throws Exception
         // load dynamic variables
         loadDynamicVariables();
     }
-
+    
+    /**
+     * updateRuntimeLocations
+     * Establish the runtime_locations.properties file based on the JBoss Application Server location
+     * (if that panel is active) or minimally just the 'runtimes' directory.  This file is read at startup
+     * and affects the runtime UI preferences.
+     */
+    private void updateRuntimeLocations()
+    {
+    	String installPath = idata.getVariable("INSTALL_PATH");
+		File folder = new File(installPath, P2DirectorStarterListener.DEVSTUDIO_LOCATION);
+
+		Properties servers = (Properties) idata.getAttribute("AS_SERVERS");
+		if (servers == null || servers.isEmpty()) {
+			servers = new Properties();
+			File runtimesFolder = new File (installPath, "runtimes");
+			RuntimePath runtimesPath = new RuntimePath(runtimesFolder.getAbsolutePath(), true);
+			servers.put("server1", runtimesPath.toString());
+		}
+		if (!folder.exists())
+			folder.mkdirs();
+
+		File file = new File(folder, "runtime_locations.properties");
+		FileOutputStream stream = null;
+		try {
```

Comment:
I was more referring to the code itself. file.createNewFile() is not required as the file will be created when writing to stream. Also for better error handling, use try (FileOutputStream = ..) as it will automatically handle errors and close the stream